aten/src/ATen/native/Vision.cpp

```diff
@@ -0,0 +1,23 @@
+#include "ATen/ATen.h"
+#include "ATen/NativeFunctions.h"
+
+#define MODE_ZEROS 0
+#define MODE_BORDER 1
+
+namespace at { namespace native {
+
+Tensor grid_sampler(const Tensor& input, const Tensor& grid, int64_t padding_mode) {
+  // as of cudnn 7102, will not work for larger than 1024
+  if (at::native::cudnn_is_acceptable(input) && padding_mode == MODE_ZEROS && input.dim() == 4 && input.size(1) <= 1024) {
+    return cudnn_grid_sampler(input, grid);
+  }
+  if (input.dim() == 4) {
```

Comment:
We should probably add a test case for input.size(1) > 1024 and input.dim() == 4 (I glanced at test_nn.py and it doesn't look like we have one).

Reply:
Good catch